crates/oxc_linter/src/rules/eslint/no_octal.rs

```diff
@@ -0,0 +1,74 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_octal_diagnostic(span: Span) -> OxcDiagnostic {
+    // See <https://oxc.rs/docs/contribute/linter/adding-rules.html#diagnostics> for details
+    OxcDiagnostic::warn("Octal literals should not be used.")
+        .with_help("Disallow octal literals.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoOctal;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow octal literals
+    ///
+    /// ### Why is this bad?
+    /// Because the leading zero which identifies an octal literal has been a source of confusion and error in JavaScript code, ECMAScript 5 deprecates the use of octal numeric literals.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// var num = 071;
+    /// var result = 5 + 07;
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// var num  = "071";
+    /// ```
+    NoOctal,
+    correctness,
+);
+
+impl Rule for NoOctal {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::NumericLiteral(numeric) = node.kind() {
+            let re = Regex::new(r"^0\d").unwrap();
```

Comment:
Use starts-with and manual char matching instead of a regular expression. These are _extremely_ costly.

Reply:
I changed it to char matching.